pkg/cmd/account/azure/create/create_test.go

```diff
@@ -0,0 +1,169 @@
+package create_test
+
+import (
+	"bytes"
+	"net/url"
+	"testing"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/OctopusDeploy/cli/pkg/cmd/account/azure/create"
+	"github.com/OctopusDeploy/cli/pkg/output"
+	"github.com/OctopusDeploy/cli/pkg/surveyext"
+	"github.com/OctopusDeploy/cli/test/fixtures"
+	"github.com/OctopusDeploy/cli/test/testutil"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/accounts"
+	octopusApiClient "github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/client"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/environments"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/resources"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+var serverUrl, _ = url.Parse("http://server")
+
+const placeholderApiKey = "API-XXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+
+var rootResource = testutil.NewRootResource()
+
+func TestAzureAccountCreatePropmtMissing(t *testing.T) {
+	const spaceID = "Space-1"
+	const envID = "Env-1"
+	_ = fixtures.NewSpace(spaceID, "testspace")
+	env := fixtures.NewEnvironment(spaceID, envID, "testenv")
+	api, qa := testutil.NewMockServerAndAsker()
+	out := &bytes.Buffer{}
+
+	opts := &create.CreateOptions{
+		CreateFlags: create.NewCreateFlags(),
+	}
+
+	errReceiver := testutil.GoBegin(func() error {
+		defer testutil.Close(api, qa)
+		octopus, _ := octopusApiClient.NewClient(testutil.NewMockHttpClientWithTransport(api), serverUrl, placeholderApiKey, "")
+		opts.Ask = qa.AsAsker()
+		opts.Octopus = octopus
+		opts.Writer = out
+		return create.PromptMissing(opts)
+	})
+
+	api.ExpectRequest(t, "GET", "/api").RespondWith(rootResource)
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Name",
+		Help:    "A short, memorable, unique name for this account.",
+	}).AnswerWith("TestAccount")
+
+	_ = qa.ExpectQuestion(t, &surveyext.OctoEditor{
+		Editor: &survey.Editor{
+			Message:  "Description",
+			Help:     "A summary explaining the use of the account to other users.",
+			FileName: "*.md",
+		},
+		Optional: true,
+	}).AnswerWith("test 123")
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Subscription ID",
+		Help:    "Your Azure subscription ID. This is a GUID in the format xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.",
+	}).AnswerWith("d2486c05-0cac-4d54-a91e-654043036f31")
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Tenant ID",
+		Help:    "Your Azure Active Directory Tenant ID. This is a GUID in the format xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.",
+	}).AnswerWith("d2486c05-0cac-4d54-a91e-654043036f32")
+
+	_ = qa.ExpectQuestion(t, &survey.Input{
+		Message: "Application ID",
+		Help:    "Your Azure Active Directory Tenant ID. This is a GUID in the format xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.",
+	}).AnswerWith("d2486c05-0cac-4d54-a91e-654043036f33")
```

Comment:
Tip: If you have some validation here (e.g. a validator that it's a GUID of the correct format) then it's very easy to throw a validation check into the middle of the test here.

```
q = qa.ExpectQuestion(t, &survey.Input { ... )

validationErr := q.AnswerWith("ddddd")
assert.EqualError(t, validationErr, "should be a guid format or something")

validationErr := q.AnswerWith("<proper guid>")
assert.Nil(t, validationErr)
```